src/gui/dialogs/addon/manager.cpp

```diff
@@ -391,6 +391,46 @@ void addon_manager::pre_show(window& window)
 	connect_signal_notify_modified(type_filter,
 		std::bind(&addon_manager::apply_filters, this));
 
+	// Language filter
+	// Prepare shown languages, source all available languages from the addons themselves
+	std::vector<std::string> languages_available;
+	for(const auto& a : addons_) {
+		for (const auto& b : a.second.locales) {
+			languages_available.push_back(b);
+		}
+	}
+	std::sort(languages_available.begin(), languages_available.end());
+	auto last = std::unique(languages_available.begin(),languages_available.end());
+	languages_available.erase(last, languages_available.end());
```

Comment:
Why are elements being erased here?

Reply:
It is for removing [duplicates](https://stackoverflow.com/questions/16476099/remove-duplicate-entries-in-a-c-vector#)

Reply:
Is there a reason to not use an std::set?

Reply:
No, I will refactor it to use set. Can I use the C++20 standard options of std::set?

Reply:
No - Wesnoth currently only allows up to C++17.